kafka-client/src/test/java/dev/responsive/kafka/integration/ResponsiveKeyValueStoreRestoreIntegrationTest.java

```diff
@@ -259,9 +259,9 @@ public void shouldRepairOffsetsIfOutOfRangeAndConfigured(final KVSchema type) th
     }
   }
 
-  @ParameterizedTest
-  @EnumSource(KVSchema.class)
```

Comment:
The fault injector doesn't actually work with `KVSchema.FACT` because it uses a different statement type. I think the off-by-one error was allowing the assertions to pass because we always reconsumed one extra record.

Reply:
I haven't found a good way to target the update from `flush` when using the `FACT` table. It uses a `DefaultBoundStatement` which is used by several other queries prior to `flush`. Without the targeted fault, the task crashes before it reads any input.